### PR TITLE
Chore/bt 159 staking

### PR DESCRIPTION
**Pull request by roy-powerfinance**
Copy pasted RC2 plug's staking module into cennznet
Changed all relevant references from pallet_staking into crml_staking

### CODE REVIEW

File: testing/Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ frame-system = { git = "https://github.com/plugblockchain/plug-blockchain", bran
 pallet-contracts = { git = "https://github.com/plugblockchain/plug-blockchain", branch = "1.0.0-rc2" }
 pallet-grandpa = { git = "https://github.com/plugblockchain/plug-blockchain", branch = "1.0.0-rc2" }
 pallet-session = { git = "https://github.com/plugblockchain/plug-blockchain", branch = "1.0.0-rc2" }
-pallet-staking = { git = "https://github.com/plugblockchain/plug-blockchain", branch = "1.0.0-rc2" }
+crml-staking = { version = "1.0.0",  path = "../crml/staking" }
```
Comment (cowboy-bebug) [inline testing/Cargo.toml]: [Optional] could move this above the `crml-transaction-payment = { ... }` line

**Review by cowboy-bebug — COMMENTED** [review]
Looks good so far! Let's wait for the ci to pass

**Review by jordy25519 — APPROVED** [review]
Does as advertised 👍

**Review by cowboy-bebug — APPROVED** [review]
LGTM!

**Pull request merged**